### PR TITLE
[MINOR][R] Fix a R style in try and finally at DataFrame.R

### PR DESCRIPTION
**Pull request by lu-wang-dl**
### What changes were proposed in this pull request?

Fix the R style issue which is not catched by the R style checker. Got error: 
```
R/DataFrame.R:1244:17: style: Closing curly-braces should always be on their own line, unless it's followed by an else.
}, finally = {
 ^
lintr checks failed.
```

### Why are the changes needed?

To fix the style issue in R.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually tested.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I think it correctly catches. Maybe the version matters in your local.

**Comment by HyukjinKwon** [comment]
What error did you see? Maybe we should consider about upgrading lint-r

**Comment by lu-wang-dl** [comment]
@HyukjinKwon Updated the error message in the description. The error is not showing in the Linters tests. Dig out what happened.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -241,7 +241,7 @@ jobs:
       run: |
         sudo apt-get install -y libcurl4-openssl-dev
         sudo Rscript -e "install.packages(c('devtools'), repos='https://cloud.r-project.org/')"
-        sudo Rscript -e "devtools::install_github('jimhester/lintr@v2.0.0')"
+        sudo Rscript -e "devtools::install_github('jimhester/lintr')"
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:244]: Looks like still 2.0.0 is the latest version. Let's revert this line

**Comment by HyukjinKwon** [comment]
@lu-wang-dl let's keep the PR template https://github.com/apache/spark/blob/master/.github/PULL_REQUEST_TEMPLATE although it's minor

**Comment by HyukjinKwon** [comment]
Which lintr version do you use btw? The fix is fine.

**Comment by HyukjinKwon** [comment]
Okay, I am going to merge this anyway since it's the right style and for some reasons linter can detect.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Pull request closed without merging**